src/Components/Shared/src/RootComponentOperation.cs

```diff
@@ -8,14 +8,8 @@ internal sealed class RootComponentOperation
     // Represents the type of root component operation to perform.
     public RootComponentOperationType Type { get; set; }
 
-    // When adding a root component, this is the selector ID
-    // to round-trip back to the client so it knows which DOM
-    // element the component should be attached to.
-    public int? SelectorId { get; set; }
-
-    // The ID of the component to use during an update or remove
-    // operation.
-    public int? ComponentId { get; set; }
+    // The client side ID of the component to perform the operation on.
+    public int SsrComponentId { get; set; }
 
```

Comment:
I should clarify that there is a concrete motivation for doing this that relates specifically to this PR.

Since components can be removed/reinitialized upon parameter updates, that means the interactive component ID can change after the browser supplies new component parameters. This would require our JS logic to avoid supplying further parameter updates until it heard back from .NET about whether the component ID changed. This is somewhat fragile, so we would want to avoid doing this as possible.

With this improvement, the browser can supply parameter updates or initiate disposal of an interactive component even before .NET sends any data back to JS. This simplified our existing logic quite a bit, since the JS code has to worry less about asynchrony, and it saved us from complicating the logic further by having to account for the interactive component ID changing dynamically after initialization.